app/script/backup/BackupRepository.js

```diff
@@ -132,45 +138,67 @@ z.backup.BackupRepository = class BackupRepository {
-      return Promise.all(
-        Object.values(archive.files)
-          .filter(zippedFile => zippedFile.name !== BackupRepository.CONFIG.META_FILENAME)
-          .map(zippedFile => zippedFile.async('string').then(value => ({content: value, filename: zippedFile.name})))
-      );
+    return this.verifyMetadata(files)
+      .then(() => this._extractHistoryFiles(files))
+      .then(fileDescriptors => this._importHistoryData(fileDescriptors, initCallback, progressCallback))
+      .catch(error => {
+        this.logger.error(`Failed to import history: ${error.message}`, error);
+        throw error;
+      });
+  }
+
+  _importHistoryData(fileDescriptors, initCallback, progressCallback) {
+    initCallback(fileDescriptors.length);
+
+    const conversationData = fileDescriptors.find(fileDescriptor => {
+      return fileDescriptor.filename === BackupRepository.CONFIG.FILENAME.CONVERSATIONS;
     });
 
-    return unzipPromise.then(fileDescriptors => {
-      initCallback(fileDescriptors.length);
+    const eventData = fileDescriptors.find(fileDescriptor => {
+      return fileDescriptor.filename === BackupRepository.CONFIG.FILENAME.EVENTS;
+    });
 
-      const importPromises = fileDescriptors.map(fileDescriptor => {
-        if (this.isCanceled) {
-          throw new z.backup.CancelError();
-        }
-        const tableName = fileDescriptor.filename.replace('.json', '');
-        const isConversationsTable = tableName === z.storage.StorageSchemata.OBJECT_STORE.CONVERSATIONS;
+    return this._importHistoryConversations(conversationData, progressCallback).then(() => {
+      return this._importHistoryEvents(eventData, progressCallback);
+    });
+  }
 
-        const entities = isConversationsTable
-          ? JSON.parse(fileDescriptor.content)
-          : JSON.parse(fileDescriptor.content).map(entity => this.mapEntityDataType(entity));
+  _importHistoryConversations(conversationData, progressCallback) {
+    if (this.isCanceled) {
+      return Promise.reject(new z.backup.CancelError());
+    }
 
-        const importPromise = isConversationsTable
-          ? this.conversationRepository.updateConversations(entities)
-          : this.backupService.importEntities(tableName, entities);
+    const entities = JSON.parse(conversationData.content).map(entity => this.mapEntityDataType(entity));
+    return this.conversationRepository.updateConversations(entities).then(() => {
+      this.logger.log(`Imported state of '${entities.length}' conversations from backup`, conversationData);
+      progressCallback();
+    });
+  }
 
-        return importPromise.then(progressCallback);
-      });
+  _importHistoryEvents(eventData, progressCallback) {
+    if (this.isCanceled) {
+      return Promise.reject(new z.backup.CancelError());
+    }
 
-      return Promise.all(importPromises);
+    const entities = JSON.parse(eventData.content);
+    return this.backupService.importEntities(this.EVENTS_STORE_NAME, entities).then(() => {
```

Comment:
I feel like this `importEntities` method is now somewhat useless since it cannot handle conversations properly.

So two options here:
- either `importEntities` is renamed `importEvents` and the semantic is now correct;
- either `importEntities` knows how to handle `conversations` and we can refactor a bit of code from the `backupRepository` (cause it won't have to branch out anymore)

Reply:
my point being, if another developer want to use this method latter on, it won't work as expected